3-CostInformation/Get-CostInformation.ps1

```diff
@@ -0,0 +1,146 @@
+<#
+.SYNOPSIS
+    Take a collection of given resource IDs and return the cost incurred during previous months,
+    grouped as needed. For this we use the Microsoft.CostManagement provider of each subscription.
+    Requires Az.CostManagement module 0.4.2 or later.
+    PS1> Install-Module -Name Az.CostManagement
+
+.PARAMETER startDate
+    The start date of the period to be examined (default is the first day of the previous month)
+
+.PARAMETER endDate
+    The end date of the period to be examined (default is the last day of the previous month)
+
+.PARAMETER resourceFile
+    A JSON file containing the resources
+
+.PARAMETER outputFile
+    The CSV file to export the results to, otherwise displayed in the console
+
+.EXAMPLE
+    .\Get-CostInformation.ps1
+    .\Get-CostInformation.ps1 -startDate "2023-01-01" -endDate "2023-06-30" -resourceFile "resources.json" -outputFile "resource_cost.csv"
+
+#>
+
+param (
+    [string]$startDate    = (Get-Date).AddMonths(-1).ToString("yyyy-MM-01"),                    # the first day of the previous month
+    [string]$endDate      = (Get-Date).AddDays(-1 * (Get-Date).Day).ToString("yyyy-MM-dd"),     # the last day of the previous month
+    [string]$resourceFile = "resources.json",
+    [string]$outputFile   = "resource_cost.csv",
+    [switch]$testMode
+)
+
+# Output to file (true) or console (false)
+$fileOutput = $true
+
+# Timeframe
+# Supported types are BillingMonthToDate, Custom, MonthToDate, TheLastBillingMonth, TheLastMonth, WeekToDate
+$timeframe = "Custom"
+
+# Granularity
+# Supported types are Daily and Monthly so far. Omit just to get the total cost.
+#$granularity = "Monthly"
+
+# Type
+# Supported types are Usage (deprecated), ActualCost, and AmortizedCost
+# https://stackoverflow.com/questions/68223909/in-the-azure-consumption-usage-details-api-what-is-the-difference-between-the-m
+$type = "AmortizedCost"
```

Comment:
This should be a parameter with a validateset to include only relevant supported types